internal/pkg/agent/application/coordinator/state/state.go

```diff
@@ -71,13 +76,24 @@ type stateSetterOpt func(ss *stateSetter)
 
 // NewCoordinatorState creates the coordinator state manager.
 func NewCoordinatorState(state agentclient.State, msg string, fleetState agentclient.State, fleetMsg string, logLevel logp.Level) *CoordinatorState {
-	return &CoordinatorState{
+	cs := &CoordinatorState{
 		state:        state,
 		message:      msg,
 		fleetState:   fleetState,
 		fleetMessage: fleetMsg,
 		logLevel:     logLevel,
+
+		// subscribeChan is synchronous: once Subscribe returns, the caller is
+		// guaranteed to be included in future state change notifications.
+		subscribeChan: make(chan StateSubscription),
+
+		// stateChangedChan is asynchronous with buffer size 1: this guarantees
+		// that state changes will propagate but multiple simultaneous changes
+		// will not accumulate.
+		stateChangedChan: make(chan struct{}, 1),
 	}
+	go cs.stateReporter()
```

Comment:
What cleans up this go routine? We should ensure that when the coordinator is stopped that this go routine is also stopped.

Reply:
That is true for the life of the Elastic Agent, but is not true for testing. I believe we should clean this up properly. That previous comment said that the goroutines that you are speaking of where cleaned up when the context was cancelled, so they where cleaned up by the caller no?